### DISCUSSION TITLE
Community Poll: Modern Icon-Pack Style for Eclipse

### DISCUSSION THREAD

**Discussion by Michael5601** [eclipse-platform/eclipse.platform.ui#2588] [Polls]
The following icon styles created by @IamLRBA and @Michael5601 are part of a new initiative for a new modern icon-pack for Eclipse.
Your feedback will help decide which style will be chosen for the icon-pack.  

Please vote for the styles that you like the most and think will work best for Eclipse. 

**How to Participate:**  
- At the bottom of the poll, you can vote for your favorite style.  
- The background colors represent Eclipse’s light and dark themes, allowing you to evaluate the icons in their intended environments.  
- Remember, this poll is about choosing an *icon style*, not the specific icons themselves. If you like a style but dislike a particular icon, you should still vote for the style since these designs are prototypes.  
- The style with the most votes will be discussed further within the community and will serve as the foundation for a new style guide.  
- For comparison, the poll also includes the current icons used in Eclipse. If you prefer the original style, feel free to vote for it as well.  

Thank you for participating!

| Number | style | Light 64x64px | Light 32x32px | Light 16x16px  | Dark 64x64px | Dark 32x32px  | Dark 16x16px |
|---------------------|--------------------|--------------------|--------------------|--------------------|--------------------|--------------------|--------------------|
| 0 | Original | ![original_light-mode_64x64](https://github.com/user-attachments/assets/e87a58ae-9758-46c9-87fd-311e50127543) | ![original_light-mode_32x32](https://github.com/user-attachments/assets/83445be5-2709-4e6a-b83c-f16540061b9f) | ![original_light-mode_16x16](https://github.com/user-attachments/assets/dab59ef2-17e6-4502-8f38-8afd28c3efb3) | ![original_dark-mode_64x64](https://github.com/user-attachments/assets/1f757b6c-8bc3-48a5-9a18-39eb70d84ed4) | ![original_dark-mode_32x32](https://github.com/user-attachments/assets/2d206a76-b208-44cf-b0ab-8f19b723bbdf) | ![original_dark-mode_16x16](https://github.com/user-attachments/assets/08d1688e-dd92-4115-aaee-6d2e1be42f29) |
| 1 | Mono  | ![mono_light-mode_64x64](https://github.com/user-attachments/assets/e178865c-68d2-433f-88b9-71b3fe69ca8e) | ![mono_light-mode_32x32](https://github.com/user-attachments/assets/fbf31f58-2f12-48a0-a0d3-da31caa949d8) | ![mono_light-mode_16x16](https://github.com/user-attachments/assets/91612f95-b637-4635-8d03-f6b79c00d416) | ![mono_dark-mode_64x64](https://github.com/user-attachments/assets/5f410713-f9a0-4ddd-9cbf-9edbc5073732) | ![mono_dark-mode_32x32](https://github.com/user-attachments/assets/e6d9a1ff-a126-4dec-b569-8e6662b759d7) | ![mono_dark-mode_16x16](https://github.com/user-attachments/assets/3a801714-6ec0-4c48-9005-734a7e9ee83d) |
| 2 | Dual-Tone  | ![dual-tone_light-mode_64x64](https://github.com/user-attachments/assets/3c9cd107-1f5f-4593-8b7f-0c8dd0ece779) | ![dual-tone_light-mode_32x32](https://github.com/user-attachments/assets/f176d1b7-8240-471c-9c20-62929270e720) | ![dual-tone_light-mode_16x16](https://github.com/user-attachments/assets/4b6b6987-dc50-45a8-b941-ddf6309ce85f) | ![dual-tone_dark-mode_64x64](https://github.com/user-attachments/assets/14f2d679-9c01-4918-bec0-78b09401c22a) | ![dual-tone_dark-mode_32x32](https://github.com/user-attachments/assets/ead9e93f-34c4-4d9e-b77c-52f576aed7aa) | ![dual-tone_dark-mode_16x16](https://github.com/user-attachments/assets/055d975a-d21b-43c0-993b-98cbc0090d67) |
| 3 | Outline 1  | ![outline_light-mode_64x64](https://github.com/user-attachments/assets/6db8667d-7bfa-4768-8164-75ec274b4cdb) | ![outline_light-mode_32x32](https://github.com/user-attachments/assets/9cb802e7-7a48-411f-bb15-69f0a3297590) | ![outline_light-mode_16x16](https://github.com/user-attachments/assets/2e56d72d-aa8c-4430-bb7b-fa875f66e19a) | ![outline_dark-mode_64x64](https://github.com/user-attachments/assets/2129af86-44bd-461f-a681-d10dfd820fa1) | ![outline_dark-mode_32x32](https://github.com/user-attachments/assets/f0303ae4-58ef-41c6-9c76-a87202b79f3b) | ![outline_dark-mode_16x16](https://github.com/user-attachments/assets/40cd3896-c6e2-4492-9a7e-e267650b10fe) |
| 4 | One-Line  | ![one-line_light-mode_64x64](https://github.com/user-attachments/assets/abcc5ead-ebcc-41a3-807a-6792f7f18660) | ![one-line_light-mode_32x32](https://github.com/user-attachments/assets/fb985456-1019-4f74-bd53-8dfbcdbc9440) | ![one-line_light-mode_16x16](https://github.com/user-attachments/assets/6c3d8b9a-4697-4ea3-8696-b673003a5110) | ![one-line_dark-mode_64x64](https://github.com/user-attachments/assets/a818824f-3b47-4675-9fa0-2e655d7ea0cf) | ![one-line_dark-mode_32x32](https://github.com/user-attachments/assets/607b1659-13cd-46a9-8780-298f3375b7d5) | ![one-line_dark-mode_16x16](https://github.com/user-attachments/assets/bb024fd5-f7e0-4428-957d-667ad9800202) |
| 5 | Outline 2  | ![outline2_light-mode_64x64](https://github.com/user-attachments/assets/f69c5651-111e-4f5a-b754-3070e322c82f) | ![outline2_light-mode_32x32](https://github.com/user-attachments/assets/84f11615-248f-496a-a2de-b3022efddd88) | ![outline2_light-mode_16x16](https://github.com/user-attachments/assets/35dd72f2-26f5-4ea6-bf1e-267d20dd78ed) | ![outline2_dark-mode_64x64](https://github.com/user-attachments/assets/849d012b-9bac-4635-909a-68fc3684ae37) | ![outline2_dark-mode_32x32](https://github.com/user-attachments/assets/7c9ab1e9-862a-4a95-b76a-223dcdf5653e) | ![outline2_dark-mode_16x16](https://github.com/user-attachments/assets/00598279-c914-434a-a20f-de753084c6eb) |
| 6 | Solid  | ![solid_light-mode_64x64](https://github.com/user-attachments/assets/7004166f-3ebf-4d2e-9f3c-85554af80b50) | ![solid_light-mode_32x32](https://github.com/user-attachments/assets/423208bd-19c2-4de8-9d3c-cabaa46e342f) | ![solid_light-mode_16x16](https://github.com/user-attachments/assets/99b85448-5580-46e6-ab83-d4d3d15823f3) | ![solid_dark-mode_64x64](https://github.com/user-attachments/assets/d6e6ba1f-4f2e-4dc4-a2c6-bee3b915cdad) | ![solid_dark-mode_32x32](https://github.com/user-attachments/assets/32c5937a-c6db-4355-aee3-e603e22be3a1) | ![solid_dark-mode_16x16](https://github.com/user-attachments/assets/b4127479-8849-427f-9b77-a05adb3fce89) |
| 7 | Abstract  | ![abstract_light-mode_64x64](https://github.com/user-attachments/assets/1c4a3388-aa34-4b57-b6d2-80bb0b801259) | ![abstract_light-mode_32x32](https://github.com/user-attachments/assets/00498266-cb83-48bb-a172-4bafeec5400a) | ![abstract_light-mode_16x16](https://github.com/user-attachments/assets/2456b0fe-31dc-40de-9f98-566bb7231446) | ![abstract_dark-mode_64x64](https://github.com/user-attachments/assets/4757b1b4-8e41-4202-8605-d0bb2375f57d) | ![abstract_dark-mode_32x32](https://github.com/user-attachments/assets/90e810dc-c44b-4abb-b3ff-b61657d03e54) | ![abstract_dark-mode_16x16](https://github.com/user-attachments/assets/7f337404-d660-4515-b1b9-cc64f4b7d6a6) |

**Comment by Claudio140** [comment]
For the outlined icons it might make sense to add colors to some of them, similar to how IntelliJ icons look:
![image](https://github.com/user-attachments/assets/b1fada1c-fdac-48e3-916d-48fe5163ffac)
![image](https://github.com/user-attachments/assets/a8422550-eca2-4e83-9ae2-b15211bec0b8)
I don't think it would be well readable if all of them were gray, especially if there are many icons close to each other like in a toolbar.

Additionally, if all icons were gray, I can't really imagine how one would easily distinguish active from disabled icon buttons.

**Reply by Michael5601** [reply]
Thank you for the feedback. The screenshot you provided also supports a disabled icon, so I don't see a problem with supporting icon enablement with monochromatic icons.

**Reply by mickaelistria** [reply]
FWIW, it's not clear to me which ones are disabled with the screenshot above. This screenshot is IMO highlighting the issue more than diminishing it.

**Reply by Michael5601** [reply]
The one pause icon in the middle is disabled. All other icons are enabled from which some are colored.

**Reply by Claudio140** [reply]
For me this is more about visual context than about single disabled buttons. Imagine all the icons were grayscale, then I think it would be harder to tell them apart.

A disabled icon like the pause button next to other gray icons can still be readable, but it is easier if there is at least a colored icon next to the disabled on one side.
Of course, other visual guides like separators and grouping in general also help to quickly identify buttons at a glance.

**Comment by mickaelistria** [comment]
I also am worried about the impossibility to express icons enablement. Currently is colored for enabled vs grayscale for disabled. What's the proposed solution for enabled vs disabled here?

**Reply by Michael5601** [reply]
We had a discussion about this [here](https://github.com/eclipse-platform/eclipse.platform.images/issues/108#issuecomment-2510704478) and my argument is that other IDEs also have modern monochromatic icons while supporting icon enablement as seen [here](https://github.com/eclipse-platform/eclipse.platform.images/issues/108#issuecomment-2510868678).

The idea would be that the disabled version of an icon is darker/brighter than the original.

**Comment by IamLRBA** [comment]
@Michael5601 Thank you for your guidance and also to the other community members thank you for the detailed discussions so far!. I’d like to suggest an approach that combines the strengths of both perspectives regarding icon distinction and readability:

1. **Subtle Accent Colors**: While maintaining a monochromatic or dual-tone theme, small accent colors could be used selectively to signify functionality or state (e.g., active/disabled, critical, or recommended actions). 

I believe this should try and work with a suggestion of @Claudio140 

> For the outlined icons it might make sense to add colors to some of them, similar to how IntelliJ icons look: ![image](https://private-user-images.githubusercontent.com/10924657/393219132-b1fada1c-fdac-48e3-916d-48fe5163ffac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4ODk0MzcsIm5iZiI6MTczMzg4OTEzNywicGF0aCI6Ii8xMDkyNDY1Ny8zOTMyMTkxMzItYjFmYWRhMWMtZmRhYy00OGUzLTkxNmQtNDhmZTUxNjNmZmFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjExVDAzNTIxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzYTNlNDA0MTQxMjM2ZDQxZGY3N2VkOGUyNWJlNzQ1MTQ4NTEzYmE3OWZmMjQyZWZlMjFmZTc2ODBkZTEwMDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ISZWoDu3RekF8SyGYfeZvsGIkv0dxroUlbLcEsu_uHc) ![image](https://private-user-images.githubusercontent.com/10924657/393219729-a8422550-eca2-4e83-9ae2-b15211bec0b8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4ODk0MzcsIm5iZiI6MTczMzg4OTEzNywicGF0aCI6Ii8xMDkyNDY1Ny8zOTMyMTk3MjktYTg0MjI1NTAtZWNhMi00ZTgzLTlhZTItYjE1MjExYmVjMGI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjExVDAzNTIxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmMDc4ZDZkOGIzNTdjNTBiNzU2NjgyNDNmMTVhMmEwYTM0MTllM2ZkM2NmZjMxNmUzYmIxOWQ3M2FlNGIzNmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.wTEw0kE_ETTI1pZpDrwZ27AkA50BuZbi13PQ1vfv_7M) I don't think it would be well readable if all of them were gray, especially if there are many icons close to each other like in a toolbar.
> 

These accents wouldn’t overwhelm the minimalist aesthetic (which is something i believe @Michael5601 is gunning for us to achieve) but could enhance visual clarity.


3. **Alternative Visual Indicators**: Alongside color differentiation for enablement states, we could also consider using subtle design tweaks like:
   - A slight opacity change or a dashed outline for disabled icons.
   - A glowing or shadow effect for active icons. 
   
This was suggested by @Michael5601 
   > 
> The idea would be that the disabled version of an icon is darker/brighter than the original.


4. **User Preferences**: Could we offer a toggle for the community to switch between strictly monochrome icons and those with subtle accents? This might cater to a broader audience while preserving the modern aesthetic.

5. **Prototypes for Testing**: Before finalizing, it might be useful to test these ideas with a small focus group to ensure usability, especially in complex UIs like toolbars or dense menus.

Ultimately, I think the goal is to balance modern aesthetics with functional clarity. This way, the icons serve their purpose effectively without compromising on the design ethos.
Thank you all again!

**Reply by Michael5601** [reply]
Thank you for your input, @IamLRBA. The community poll provided some valuable insights, even though the results were quite balanced except for the dual-tone style. This highlights how subjective design preferences can be. 

I apologize for my limited time right now, but I’ll try to push this project forward so we can finalize the style guide and implement a quality control plan that enables independent icon submissions. I really appreciate your support with this project. 😊

With the poll finished, we can now move to the next phase: refining the most suitable style. I’ll create a new issue to discuss this. 

Here are my thoughts on the "winning" icon styles:  
1. **One-Line Style**: It received more votes than expected, but its abstract style makes it difficult to design complex icons (e.g., editor window + pin). While interesting for this poll, I wouldn’t recommend it as a base for the new style guide.  
2. **Outline 1 Style**: I heard many concerns about icon enablement. The small surface are of Outline 1 style makes disabled icons less distinguishable compared to Mono style. Mono, with its larger surface area, is better suited for this. I agree that the icon enablement is harder to provide for Outline 1 but I don't agree for Mono as the surface area is large. 
3. **Dual-Tone Style**: This style received the most votes. A style like Mono (Dual-Tone is essentially Mono with more color), with features like round corners, large surface areas, and bold 2px lines, supports dual-tone designs and icon enablement.  

That said, including the dual-tone style may have influenced the results. It’s unclear how many would have favored Mono without dual-tone as an option. However, since it aligns well with usability and dual-tone integration, I suggest focusing on refining Mono-style aspects. If anyone disagrees, please share your thoughts in the upcoming issue.  

Also here is some feedback on your comment, @IamLRBA:  
1. **Subtle Accent Colors**: This is hard to visualize without seeing what exactly you mean. For me the dual-tone style does what you described here. If you have an idea that we can consider implementing into the new style it would be great if you can provide simple sketches. 
2. **Alternative Indicators**: On the technical side we are right now bound to only using color as indicator for icon-enablement. I am implementing a way to dynamically alter SVG at runtime which allows more automatic customization but if an Eclipse RCP product doesn't use SVGs this wouldn't work. Even though I like the idea of additional indicators there are some technical bounds to this right now. 
3. **User Preferences**: This means either we design all icons in both variants which results in not only one icon pack but in two. Or we dynamically change the icons at runtime which is not trivial and will only properly work for SVGs. As we don't even have the technical base for icon theming yet I think we should avoid adding more requirements.
5. **Prototypes for Testing**: Testing icons in context before finalizing is a good idea. Once we’ve designed a few icons based on the new style, we can load them locally into Eclipse and share screenshots for evaluation. This feedback can refine the style guide further.  

Thank you again for your suggestions and for supporting this project.

**Reply by Michael5601** [reply]
See this [issue](https://github.com/eclipse-platform/eclipse.platform.images/issues/117) for further discussion.

**Comment by IamLRBA** [comment]
Thank you, @Michael5601, for your thoughtful and detailed response. I appreciate the time and effort you’re putting into this project despite your limited availability. 😊

Regarding your feedback on the suggestions I provided:

You’re right that maintaining multiple icon packs or dynamic runtime changes could complicate the implementation significantly. Perhaps, as a long-term vision, this idea could complement an eventual icon theming system. For now, prioritizing a single, robust style makes sense.
I’m glad to hear that testing icons in context is part of the plan. This will undoubtedly provide valuable feedback to refine the style guide further. I’m happy to participate in testing or contribute ideas to ensure the icons function well in real-world scenarios.

Lastly, I agree that the Mono/Dual-Tone styles are strong candidates due to their balance of modern aesthetics and functionality. Refining these styles, as you suggested, seems like the best path forward.

Thank you again for engaging with the feedback so thoughtfully. I look forward to seeing how the style evolves in the next phase!

**Comment by laeubi** [comment]
Please be aware that we have one very hard to takle problem to solve: **Extensions** ... and Eclipse heavily depends on them. So If Eclipse switches to whatever "style" it immediately will lead to a wild mixture of old/new styles. So who will take care to update all these as well? Another one is about custom RCP applications, or even supporting "older" Eclipse releases ...

**Reply by Michael5601** [reply]
This is part of the icon theming implementation. There is already a [PR](https://github.com/eclipse-platform/eclipse.platform.ui/pull/14) that was worked on 1.5 years ago. A few days ago the work on the PR started again. As far as I know the icon theming shall also work for all bundles/extensions. 

The new style is planned as a new icon pack which can be optionally activated with the mentioned icon theming. This means custom RCP applications are not obligated to use the new style but they can if they want. Older Eclipse releases will just use the current default theme and won't have access to the new style. 

Plugins from the market place that provide new custom icons are out of scope for this initiative. Providers of the plugins need to update their icons regarding the icon theming or users have to endure mixed icons.
